src/Storage/StoredMessageRepository.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KaroIO\MessengerMonitorBundle\Storage;
+
+/**
+ * @internal
+ */
+final class StoredMessageRepository
+{
+    private $doctrineConnection;
+    private $tableName;
+
+    public function __construct(DoctrineConnection $doctrineConnection, string $tableName)
+    {
+        $this->doctrineConnection = $doctrineConnection;
+        $this->tableName = $tableName;
+    }
+
+    public function saveMessage(StoredMessage $storedMessage): void
+    {
+        $this->doctrineConnection->executeQuery(
+            <<<SQL
+INSERT INTO {$this->tableName}
+(id, class, dispatched_at)
+VALUES (:id, :class, :dispatched_at)
+SQL
+            ,
+            [
+                'id' => $storedMessage->getId(),
+                'class' => $storedMessage->getMessageClass(),
+                'dispatched_at' => $storedMessage->getDispatchedAt()->format('Y-m-d H:i:s'),
+            ]
+        );
```

Comment:
Since we have the `DoctrineConnection` class, it might be better to encapsulate all the logic about the table inside that class - e.g. something like `$this->doctrineConnection->insertMessage($storedMessage->getId(), $storedMessage->getMessageClass(), ...` - and same for update and find. The fact that the `$tableName` is injected into both classes is what "struck" me as not ideal.

I also think that we need to use the query builder instead of writing raw SQL - like https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php#L120-L128

I'm just guessing - but I think that is what will cause correct SQL to be generated based on the user's actual platform - i.e. different SQL for MySQL vs pgSQL